controllers/envoy/secrets_controller.go

```diff
@@ -103,8 +105,37 @@ func (r *SecretReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 	return reconcile.Result{}, nil
 }
 
+func filterTLSTypeCertificatesPredicate() predicate.Predicate {
+	return predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
+			switch o := e.Object.(type) {
+			case *corev1.Secret:
+				if o.Type == "kubernetes.io/tls" {
+					return true
+				}
+				return false
+
+			default:
+				return true
+			}
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			switch o := e.ObjectNew.(type) {
+			case *corev1.Secret:
+				if o.Type == "kubernetes.io/tls" {
+					return true
+				}
+				return false
+			default:
+				return true
+			}
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool { return false },
+	}
+}
+
```

Comment:
This is done in order to filter out some events so the reconciled is not triggered so often. It is important because it is watching any Secret in the cluster, so this way at least we narrow it down to just Secrets of type `kubernetes.io/tls` and just for create and update operations.